lib/vtls/schannel.c

```diff
@@ -708,22 +727,235 @@ schannel_acquire_credential_handle(struct Curl_easy *data,
   if(!backend->cred) {
     failf(data, "schannel: unable to allocate memory");
 
+#ifdef HAS_CLIENT_CERT_PATH
     if(client_certs[0])
       CertFreeCertificateContext(client_certs[0]);
+#endif
 
     return CURLE_OUT_OF_MEMORY;
   }
   backend->cred->refcount = 1;
 
-  sspi_status =
-    s_pSecFn->AcquireCredentialsHandle(NULL, (TCHAR *)UNISP_NAME,
-                                       SECPKG_CRED_OUTBOUND, NULL,
-                                       &schannel_cred, NULL, NULL,
-                                       &backend->cred->cred_handle,
-                                       &backend->cred->time_stamp);
+  /* Windows 10, 1809 (a.k.a. Windows 10 build 17763) */
+  if(curlx_verify_windows_version(10, 0, 17763, PLATFORM_WINNT,
+      VERSION_GREATER_THAN_EQUAL)) {
+
+    int disable_aes_gcm_sha384 = 0;
+    int disable_aes_gcm_sha256 = 0;
+    int disable_chacha_poly = 1;
```

Comment:
How about using `bool` and `TRUE`/`FALSE` for these?

Reply:
I couldn't remember if c89 had the `bool` type (I do most of my programming in C++20). But you're right, I'll change this.

Reply:
It doesn't, but we provide our own! :grin: